File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    answer = 0
    
    bridge = deque([0]*bridge_length)
    bridge_weight = 0
    
    idx = 0
    time = 0
    while(bridge):
        bridge_weight -= bridge.popleft()
        time += 1
        if idx < len(truck_weights):
            if bridge_weight + truck_weights[idx] <= weight:
                bridge.append(truck_weights[idx])
                bridge_weight+=truck_weights[idx]
                idx+=1
            else:
                bridge.append(0)

    return time

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    var answer = 0;
    let bridge = new Array(bridge_length).fill(0)
    let bridge_weight = 0 
    
    let idx = 0
    while(bridge.length){
        answer++
        bridge_weight -= bridge.shift()
        
        if (idx < truck_weights.length){
            if (truck_weights[idx] + bridge_weight <=weight){
                bridge.push(truck_weights[idx])
                bridge_weight+=truck_weights[idx]
                idx++
            }
            else bridge.push(0)
        }
        
    }
    
    
    
    return answer;
}